launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +121,30 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   *
+   * @since 2.2.0
+   * @return This launcher.
+   */
+  public SparkLauncher stopIfLauncherShutdown() {
+    this.stopIfLauncherShutdown = true;
+    return this;
+  }
+
+  /**
+   * Specifies that Spark Submit be launched as a daemon thread using reflection.
```

Comment:
Remove "using reflection". The user doesn't really need to know how things are implemented. Instead, you should mention that the necessary Spark artifacts need to be available in the application's classpath.